Objects/memoryobject.c

```diff
@@ -3398,6 +3412,53 @@ PyTypeObject _PyMemoryIter_Type = {
     .tp_iternext = memoryiter_next,
 };
 
+/**************************************************************************/
+/*                      Memoryview Reversed Iterator                      */
+/**************************************************************************/
+
+static PyObject *
+memorview_reverse_iternext(PyObject *self)
+{
+    memoryiterobject *it = _PyMemoryViewIter_CAST(self);
+    PyMemoryViewObject *seq = it->it_seq;
+    if (seq == NULL) {
+        return NULL;
+    }
+    if (it->it_index >= 0 && it->it_index < it->it_length) {
+        // FEAT(picnixz): can we omit "it->it_index < it->it_length"?
+        return memoryiter_iter_nth(seq, it->it_index--, it->it_fmt);
+    }
+    it->it_seq = NULL;
+    Py_DECREF(seq);
+    return NULL;
+}
+
+/*[clinic input]
+memoryview.__reversed__
+
+Return a reverse iterator over this memory view.
+[clinic start generated code]*/
+
+static PyObject *
+memoryview___reversed___impl(PyMemoryViewObject *self)
+/*[clinic end generated code: output=25f9b4345f4720ad input=3c35e9267ad7fcc6]*/
+{
+    return memoryiter_new((PyObject *)self, 1);
+}
+
+
+PyTypeObject _PyMemoryRevIter_Type = {
+    PyVarObject_HEAD_INIT(&PyType_Type, 0)
+    "memory_reverseiterator",
+    sizeof(memoryiterobject),
+    .tp_dealloc = memoryiter_dealloc,
+    .tp_getattro = PyObject_GenericGetAttr,
```

Comment:
Isn't this set by default?

Reply:
I actually copied what the `memory_iterator` did. Since I didn't want to have a disparity between the normal and the reverse iterator, I decided to keep it in both.